packages/studio-ui/src/components/AddPageButton/LayoutSelector.tsx

```diff
@@ -0,0 +1,68 @@
+import { ChangeEvent, useCallback, useState } from "react";
+import { FlowStepModalProps } from "./FlowStep";
+import useStudioStore from "../../store/useStudioStore";
+import DialogModal from "../common/DialogModal";
+import { selectCssClasses } from "../UnionPropInput";
+import { twMerge } from "tailwind-merge";
+
+export default function LayoutSelector({
+  isOpen,
+  handleClose,
+  handleConfirm,
+}: FlowStepModalProps) {
+  const layouts = useStudioStore(
+    (store) => store.layouts.layoutNameToLayoutState
+  );
+  const [selectedLayout, setSelectedLayout] = useState<string>("");
+  const [errorMessage, setErrorMessage] = useState<string>();
+
+  const onConfirm = useCallback(async () => {
+    const layoutState = layouts[selectedLayout];
+
+    try {
+      await handleConfirm(layoutState);
+      await handleClose();
```

Comment:
Can either of these methods throw an exception? Do we need them in a try-catch?

Reply:
In our `handleConfirm`, we call create page which can throw errors if something goes wrong-- we then post the error on the modal instead of closing it.  Right now, we do checks for a `getPathValue` in PagesJS repos, and a check for invalid page names.